src/celeritas/optical/ScintillationParams.hh

```diff
@@ -27,22 +28,37 @@ class ScintillationParams final : public ParamsDataInterface<ScintillationData>
   public:
     //!@{
     //! \name Type aliases
+    using SPConstParticles = std::shared_ptr<ParticleParams const>;
     using ScintillationDataCRef = HostCRef<ScintillationData>;
     //!@}
 
-    //! Material-dependent scintillation data, indexed by \c OpticalMaterialId
+    //! Scintillation data for all materials
     struct Input
     {
-        std::vector<ImportScintData> data;
+        using VecOptMatId = std::vector<OpticalMaterialId>;
+        using VecSPId = std::vector<ScintillationParticleId>;
+
+        VecOptMatId matid_to_optmatid;  //!< MaterialId to OpticalMaterialId
+        VecSPId pid_to_scintpid;  //!< ParticleId to ScintillationParticleId
+        std::vector<ImportScintData> data;  //!< Indexed by OpticalMaterialId
+        bool scintillation_by_particle;  //!< Particle or material sampling
+
+        //! Whether all data are assigned and valid
+        explicit operator bool() const
+        {
+            return !data.empty()
+                   && (!matid_to_optmatid.empty() || !pid_to_scintpid.empty());
```

Comment:
Same thing about one or the other: it shouldn't be both?

Reply:
Here in the params is where we break from the import: During import, we load whatever is in the geometry. Then, at startup time, our `ScintillationParams` is constructed based on the user-defined input and only loads one or another. This is decided via the `ScintillationParams::Input::scintillation_by_particle` boolean -- which could be also be an enum, now that I think I about it. Something like this would probably look better:
```cpp
enum class ScintillationType
{
    material,
    particle
};
```

Reply:
Yeah, either that or (better) a `std::variant` so the only data members that are truly available are the ones that correspond to the given option.